biz.aQute.repository/src/aQute/bnd/repository/p2/provider/P2Indexer.java

```diff
@@ -131,10 +138,29 @@ private Repository readRepository() throws Exception {
 		List<Artifact> artifacts = p2.getArtifacts();
 		List<Promise<Resource>> fetched = new ArrayList<>(artifacts.size());
 		Set<URI> visited = new HashSet<>(artifacts.size());
+		Set<String> visitedHash = new HashSet<>(artifacts.size());
+		Map<String,Resource> existing = new HashMap<>();
+
+		if (getBridge() != null) {
+			Requirement requirement = new RequirementBuilder(P2_CAPABILITY_NAMESPACE).addAttribute(MD5_ATTRIBUTE, "*")
```

Comment:
This is a little awkward since the addAttribute throws checked exceptions. Either everything has to be bogged down with unnecessary safety logic or we just rethrow as an assertionerror in the static initialiser block, since nothing should ever go wrong. I'll do the latter for now, please advise if this isn't okay.